src/core/custom-elements/index.js

```diff
@@ -0,0 +1,41 @@
+// @ts-check
+/**
+ * Registers custom elements and waits for them to finish their processing.
+ *
+ * Every custom element file exports:
+ * - `name`: registered name of the custom element, prefixed with `rs-`.
+ * - `default`: class defintion of the custom element.
+ * - `is`: an optional string if element extends some built-in element.
+ *
+ * Every custom element must dispatch a CustomEvent 'done' that tells the
+ * element has finished its processing, with or without errors.
+ *
+ * @typedef {{ name: string, default: Function, is?: string }} CustomElementDfn
+ */
+
+import * as changelog from "./rs-changelog.js";
+/** @type {CustomElementDfn[]} */
+const CUSTOM_ELEMENTS = [changelog];
+
+export const name = "core/custom-elements";
+
+export async function run() {
+  // prepare and register elements
+  CUSTOM_ELEMENTS.forEach(el => {
+    customElements.define(
+      el.name,
+      el.default,
+      el.is ? { extends: el.is } : undefined
```

Comment:
There is no `is` now, do you have an immediate plan to implement one? If not how about deferring this until we actually need it?

Reply:
No yet, we don't. Lets defer this until needed.